src/io/GeantImporter.cc

```diff
@@ -0,0 +1,152 @@
+}
+
+GeantImporter::~GeantImporter() = default;
+
+//---------------------------------------------------------------------------//
+/*!
+ * Operators ()
+ */
+geant_importer_data GeantImporter::operator()()
+{
+    geant_importer_data geant_data;
+    load_particle_data();
+    geant_data.particle_params = get_ParticleParams();
+
+    // Placeholder
+    // load_physics_tables();
+    // geant_data.physics_tables = get_PhysicsTables();
+
+    return geant_data;
+}
+
+geant_importer_data
+GeantImporter::operator()(std::vector<PDGNumber> pdg_numbers)
+{
+    geant_importer_data              geant_data;
+    ParticleParams::VecAnnotatedDefs defs;
+    load_particle_data();
+
+    for (PDGNumber pdg : pdg_numbers)
+    {
+        for (GeantParticle particle : this->particle_vector_)
+        {
+            if (particle.pdg == pdg.get())
+            {
+                defs.push_back(
+                    // ParticleMd
+                    {{particle.name, PDGNumber{particle.pdg}},
+                     // ParticleDef
+                     {particle.mass,
+                      particle.charge,
+                      (particle.is_stable ? ParticleDef::stable_decay_constant()
+                                          : 1. / particle.lifetime)}});
+                break;
+            }
+        }
+    }
+    geant_data.particle_params
+        = std::make_shared<ParticleParams>(std::move(defs));
+
+    /*!
+     * A similar routine for fetching only the expected physics tables
+     * should be added here
+     */
+
+    return geant_data;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Fills the vector<GeantParticle> particle_vector_
+ */
+void GeantImporter::load_particle_data()
+{
+    // Avoiding duplicates in case the method is called more than once
+    this->particle_vector_.clear();
+
+    build_object_list("particles");
+
+    TTree* tree_particles = (TTree*)this->root_input_->Get("particles");
+
+    for (auto branch_name : this->object_list_)
+    {
+        GeantParticle* particle = new GeantParticle;
+
+        tree_particles->SetBranchAddress(branch_name.c_str(), &particle);
+        tree_particles->GetEntry(0);
+
+        this->particle_vector_.push_back(*particle);
+    }
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Creates a shared_ptr<ParticleParams> with all GeantParticle objects
+ */
+std::shared_ptr<ParticleParams> GeantImporter::get_ParticleParams()
+{
+    ParticleParams::VecAnnotatedDefs defs;
+
+    for (auto particle : this->particle_vector_)
+    {
+        defs.push_back(
+            // ParticleMd
+            {{particle.name, PDGNumber{particle.pdg}},
+             // ParticleDef
+             {particle.mass,
+              particle.charge,
+              (particle.is_stable ? ParticleDef::stable_decay_constant()
+                                  : 1. / particle.lifetime)}});
```

Comment:
Instead of the inline comments, this would be more readable by replacing
```
// ParticleMd
             {{particle.name, PDGNumber{particle.pdg}},
```
with

```
ParticleMd{{particle.name, PDGNumber{particle.pdg}},
```

or even breaking those up into multiple lines (especially the decay constant) and assigning them as data members rather than in an initializer:
```c++
ParticleMd md;
md.name = particle.name;
md.pdg = PDGNumber{particle.pdg};
ParticleDef def;
def.mass = particle.mass;
...
```

Reply:
You might have missed my previous comment https://github.com/celeritas-project/celeritas/pull/30#discussion_r469990094 ?

Reply:
Yep. Missed that comment. It's fixed now.